controllers/metal3machine_controller_test.go

```diff
@@ -85,19 +84,20 @@ func setReconcileNormalExpectations(ctrl *gomock.Controller,
 		if tc.AssociateFails {
 			m.EXPECT().Associate(context.TODO()).Return(errors.New("Failed"))
 			m.EXPECT().GetBaremetalHostID(context.TODO()).MaxTimes(0)
-			m.EXPECT().Update(context.TODO()).MaxTimes(0)
 			return m
 		} else {
 			m.EXPECT().Associate(context.TODO()).Return(nil)
+			m.EXPECT().Update(context.TODO()).MaxTimes(0)
 		}
+	} else {
+		m.EXPECT().Update(context.TODO()).MaxTimes(2)
```

Comment:
why max times 2 ? if it should be called, remove the MaxTimes, it will ensure that it is called at least once.

Reply:
Fixed